protocol/linting.md

```diff
@@ -0,0 +1,16 @@
+# Proposed linting standards
+
+Linting is a best practice for developers in general and an essential aid for developers working together on open source projects. Here are kmcculloch's proposed linting standards for EDGI projects:
+
+1. All projects that constitute an application codebase (rather than a set of documents or stand-alone scripts) should adopt a **linting tool** and a **linting standard** that is appropriate for the language or framework.
```

Comment:
Yeah, wow, that's a long doc. Linters actually reduce the need for such extensive code style documentation. You don't need to explicitly state that you should always terminate your js statements with a semicolon, because the linter will fuss at you if you don't. eslint shows the machine name of each rule violation, which you can easily Google for an explanation and examples of how to fix.

The other thing to bear in mind is that there's no single standard for a given language; it varies a bit from framework to framework. In the PHP world, the general standard is four-space indents, but the Drupal standard for PHP is two-space indents. My PHP linter is smart enough to know the context of a PHP file, so I don't need to remember picky little differences like that. Which is awesome: I get very clean, uniform code with almost zero cognitive effort. If you want to include a long style guide you'll need to include a different one for each framework. It's a lot of documentation to maintain when linters are, by their very nature, self-documenting.